src/main/java/uk/gov/justice/digital/delius/service/RequirementService.java

```diff
@@ -78,16 +81,22 @@ public PssRequirements getPssRequirementsByConvictionId(String crn, Long convict
     }
 
     private Event getEvent(String crn, Long convictionId) {
-        var offender = getOffender(crn);
-        return eventRepository.findByOffenderId(offender.getOffenderId())
+        var offenderId = getOffenderId(crn);
+        return eventRepository.findByOffenderId(offenderId)
                 .stream()
                 .filter(event -> convictionId.equals(event.getEventId()))
                 .findAny()
                 .orElseThrow(() ->  new NotFoundException(format("Conviction with convictionId '%s' not found", convictionId)));
     }
 
-    private Offender getOffender(String crn) {
-        return offenderRepository.findByCrn(crn)
+    private Event getActiveEvent(String crn, Long convictionId) {
```

Comment:
This used to be like the method above, which seems a wasteful way to get an Event which we have the ID of